pkg/client/cache/listers.go

```diff
@@ -109,48 +110,56 @@ func (s *StoreToNodeLister) List() (machines api.NodeList, err error) {
 }
 
 // NodeCondition returns a storeToNodeConditionLister
-func (s *StoreToNodeLister) NodeCondition(conditionType api.NodeConditionType, conditionStatus api.ConditionStatus) storeToNodeConditionLister {
+func (s *StoreToNodeLister) NodeCondition(nodeConditionPairs ...NodeConditionPair) storeToNodeConditionLister {
 	// TODO: Move this filtering server side. Currently our selectors don't facilitate searching through a list so we
 	// have the reflector filter out the Unschedulable field and sift through node conditions in the lister.
-	return storeToNodeConditionLister{s.Store, conditionType, conditionStatus}
+	return storeToNodeConditionLister{s.Store, nodeConditionPairs}
+}
+
+type NodeConditionPair struct {
+	ConditionType   api.NodeConditionType
+	ConditionStatus api.ConditionStatus
 }
 
 // storeToNodeConditionLister filters and returns nodes matching the given type and status from the store.
 type storeToNodeConditionLister struct {
-	store           Store
-	conditionType   api.NodeConditionType
-	conditionStatus api.ConditionStatus
+	store         Store
+	condtionPairs []NodeConditionPair
 }
 
 // List returns a list of nodes that match the condition type/status in the storeToNodeConditionLister.
 func (s storeToNodeConditionLister) List() (nodes api.NodeList, err error) {
 	for _, m := range s.store.List() {
 		node := *m.(*api.Node)
-
-		// We currently only use a conditionType of "Ready". If the kubelet doesn't
-		// periodically report the status of a node, the nodecontroller sets its
-		// ConditionStatus to "Unknown". If the kubelet thinks a node is unhealthy
-		// it can (in theory) set its ConditionStatus to "False".
-		var nodeCondition *api.NodeCondition
-
-		// Get the last condition of the required type
-		for _, cond := range node.Status.Conditions {
-			if cond.Type == s.conditionType {
-				nodeCondition = &cond
-			} else {
-				glog.V(4).Infof("Ignoring condition type %v for node %v", cond.Type, node.Name)
+		matches := true
+		for _, c := range s.condtionPairs {
+			// We currently only use a conditionType of "Ready". If the kubelet doesn't
```

Comment:
This comment is now out of date, because we use two condition types as of this change.